src/bin/rational.rs

```diff
@@ -479,6 +601,33 @@ impl SynthLanguage for Math {
 
             let lvec = Self::eval_pattern(lhs, &env, n);
             let rvec = Self::eval_pattern(rhs, &env, n);
+
+            // if we are writing cvecs to a file
+            if synth.params.write_ces {
+                let mut file = OpenOptions::new()
+                    .write(true)
+                    .append(true)
+                    .open("counterexamples.txt")
```

Comment:
not txt, i think json right?

Reply:
also note that you dont need both write and append. If setting both to true, i think only append is needed.

Reply:
More importantly this wont work unless the file exists. I believe you want `create`, not `open`

Reply:
```

                let mut file = OpenOptions::new()
                    .append(true)
                    .create(true)
                    .open("counterexamples.txt")
                    .unwrap();
```

Reply:
Yeah, it should be .json. Does create only create the file if it does not exist already? I think we want to be able to build up results over different runs, so recreating the file each time may not be ideal.

Reply:
yes I believe it will do the right thing. The current version will crash because open wants the file to exist already but create doesnt. Do double check though!